velox/dwio/common/tests/DataSinkTest.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/dwio/common/DataSink.h"
+#include "velox/common/base/Fs.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+using namespace ::testing;
+using namespace facebook::velox::exec::test;
+
+namespace facebook::velox::dwio::common {
+
+TEST(LocalFileSinkTests, create) {
+  LocalFileSink::registerFactory();
+
+  auto root = fs::path(TempDirectoryPath::create()->path);
+  auto subdir = fs::path();
+  for (auto i = 0; i < 4; i++) {
+    subdir =
+        subdir / boost::uuids::to_string(boost::uuids::random_generator()());
+  }
+  auto fileName = boost::uuids::to_string(boost::uuids::random_generator()());
+
+  auto filePath = root / subdir / fileName;
+
+  auto localFileSink =
+      DataSink::create(fmt::format("file:{}", filePath.string()));
+  localFileSink->close();
+
+  EXPECT_TRUE(fs::exists(filePath.string()));
```

Comment:
Can we make sure the file is deleted at the end of the test?

Reply:
The root is created by `TempDirectoryPath::create()`, and I see the destructor of TempDirectoryPath is like

https://github.com/facebookincubator/velox/blob/05ad3d06ab7cd5fda48a87e78bf3f6d0d8d815f2/velox/exec/tests/utils/TempDirectoryPath.cpp#L30-L32

Also TempDirectoryPath::create() calls mkdtemp() underlying. So my understanding is the root directory would be deleted when the variable leaves its scope.

Reply:
Revised, and checked the directory gets deleted after the unit test finishes.

I see what you mean, and you're right. As I didn't store in a variable TempDirectoryPath::create() but stored ->path, the destructor ~TempDirectoryPath() is called right away after this line. So no directory deletion will be called after the test finishes.